source/slang/slang-emit-cuda.cpp

```diff
@@ -238,6 +238,22 @@ SlangResult CUDASourceEmitter::calcTypeName(IRType* type, CodeGenTarget target,
         }
     }
 
+    if (auto untypedBufferType = as<IRUntypedBufferResourceType>(type)) {
+        switch (untypedBufferType->getOp())
+        {
+            case kIROp_RaytracingAccelerationStructureType:
+            {
+                m_writer->emit("OptixTraversableHandle");
+                return SLANG_OK;
+                break;
+            }
+
+            default:
+                SLANG_DIAGNOSE_UNEXPECTED(getSink(), SourceLoc(), "unhandled buffer type");
+                break;
```

Comment:
You might need/want to let the cases that you don't handle here fall through to `Super::calcTypeName` below. There are other examples of untyped buffer resources that we'd want to let be handled using our ordinary CUDA translation logic (e.g., `ByteAddressBuffer` and `RWByteAddressBuffer`). Maybe those are just not implemented, though: I'm not sure.

Reply:
Makes sense. Just pushed a commit to allow unhandled cases to fall back to `Super::calcTypeName`.